### PR TITLE
nf-tests for bedtools

### LINKED ISSUE #4447: port bedtools module tests from pytest to nf-test
### Is your feature request related to a problem? Please describe

tests in https://github.com/nf-core/modules/tree/master/tests/modules/nf-core/bedtools should be ported to nf-test

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ahvigil**
## PR checklist

Implements tests in nf-test for
- `bedtools bamtobed`
- `bedtools closest`
- `bedtools complement`
- `bedtools coverage`
- `bedtools genomecov`

Closes #4447 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bedtools/genomecov/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                // extension
+                input[2] = "txt"
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
```
Comment (sateeshperi) [inline modules/nf-core/bedtools/genomecov/tests/main.nf.test:32]:
```suggestion
                { assert snapshot(process.out).match("no_scale") }
```

File: modules/nf-core/bedtools/genomecov/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                // extension
+                input[2] = 'txt'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
```
Comment (sateeshperi) [inline modules/nf-core/bedtools/genomecov/tests/main.nf.test:58]:
```suggestion
                { assert snapshot(process.out).match("dummy_sizes") }
```

File: modules/nf-core/bedtools/genomecov/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
```
Comment (sateeshperi) [inline modules/nf-core/bedtools/genomecov/tests/main.nf.test:84]:
```suggestion
                { assert snapshot(process.out).match("scale") }
```

**Review by sateeshperi — COMMENTED** [review]
minor change....when having multiple tests in a suite...better to give unique identifier names for each test in the snap

**Review by nvnieuwk — COMMENTED** [review]
Good job! Can you also add a stub test for the modules that have a stub?

**Comment by ahvigil** [comment]
> Good job! Can you also add a stub test for the modules that have a stub?

added stub tests for `bedtools bamtobed` and `bedtools genomecov`

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**